### PR TITLE
Overall code improvements

### PR DESCRIPTION
**Pull request by kpostekk**
Brief into changes:
 -  DO NOT USE `as const` (it's not recommended to use reserved keywords)
     - Fix: Use type to enforce type safety on all fields (instead of one)
 - Use crypto lib instead of package md5
     - Fix: Use createHash('md5')
 - Use `import dotenv/config` instead importing config as exported function
 - Use *if not* to ensure location header exists (using non-null assertion without comment is dangerous)
 - Set TZ in Dockerfile instead in runtime to allow TZ overwrite
 - Minor changes to .*ignore files

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -2,3 +2,9 @@ data
 logs
 .env
 dist
+node_modules/
```
Comment (bibixx) [inline .dockerignore:5]: Don't know how I have missed adding `node_modules` to `.ingore` files 🤦🏻

File: src/utils/scrape/login.ts [hunk 2]
```diff
@@ -13,16 +13,17 @@ export const login = async (cookieJar: CookieJar) => {
...
     body: urlencoded,
-    redirect: 'manual' as const
+    redirect: 'manual'
   };
 
   const firstResponse = await fetch("https://planzajec.pjwstk.edu.pl/Logowanie.aspx", requestOptions)
   const [firstCookie] = firstResponse.headers.raw()['set-cookie'][0].split(';')
-  const nextLocation = firstResponse.headers.get("location")!
+  const nextLocation = firstResponse.headers.get("location")
+  if (!nextLocation) throw new Error('Missing location header!')
```
Comment (bibixx) [inline src/utils/scrape/login.ts]: Just a styling preference

```suggestion
  const nextLocation = firstResponse.headers.get("location")

  if (!nextLocation) {
    throw new Error('Missing location header!')
  }
```

**Review by bibixx — CHANGES_REQUESTED** [review]
Thank you so much for the contribution!
Left one small comment. Otherwise we're good to go! 🚀

File: src/utils/scrape/login.ts (hunk 2 above)
Comment (kpostekk) [reply src/utils/scrape/login.ts]: I see curly barces gang {😎}

**Review by bibixx — APPROVED** [review]

File: .dockerignore (hunk 1 above)
Comment (kpostekk) [reply .dockerignore:5]: In my projects i use collection of those ignore files:
https://github.com/github/gitignore
Maybe it will be useful in future

**Pull request merged**